### PR TITLE
finished reviews db

### PR DESCRIPTION
**Pull request by Scarlette-O**
Finished db for reviews let me know if i should make any changes.

### CODE REVIEW

**Review by Vimaser — APPROVED** [review]
Looks good!

File: db/models/reviews.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const client = require("../client");
+
+async function createReview({ productId, reviewName, reviewerEmail, reviewDate }) {
+    try {
+        const { rows } = await client.query(`
+        INSERT INTO Reviews (ProductId, ReviewName, reviewerEmail, reviewDate)
```
Comment (cmceachin5888) [inline db/models/reviews.js:6]: dont use camel case here. 

also, review the table column names. you are not following the table we initially had.

CREATE TABLE product_reviews (
  id SERIAL PRIMARY KEY,
  product_id UUID NOT NULL,
  user_id INT NOT NULL,
  rating INT NOT NULL,
  review_text TEXT,
  review_date TIMESTAMP NOT NULL,
  FOREIGN KEY (product_id) REFERENCES products(id),
  FOREIGN KEY (user_id) REFERENCES users(id)
);

File: db/models/reviews.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (error) {
+        throw new Error('Error fetching reviews: ' + error.message);
+    }
+}
+
+async function getReviewsByProduct(productId) {
+    try {
+        const { rows } = await client.query(`
+        SELECT * 
+        FROM Reviews
```
Comment (cmceachin5888) [inline db/models/reviews.js:36]: should not be capitalized for "reviews"

File: db/models/reviews.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        throw new Error('Error fetching reviews: ' + error.message);
+    }
+}
+
+async function getReviewsByProduct(productId) {
+    try {
+        const { rows } = await client.query(`
+        SELECT * 
+        FROM Reviews 
+        WHERE ProductID = $1
```
Comment (cmceachin5888) [inline db/models/reviews.js:37]: dont use camel case here

**Review by cmceachin5888 — CHANGES_REQUESTED** [review]

File: db/models/reviews.js (hunk 1 above)
Comment (Scarlette-O) [reply db/models/reviews.js]: My bad I didnt even think to check if tables were already made. Will update and make changes asap.

File: db/models/reviews.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+const client = require("../client");
+
+async function createReview({ id, productId, userId, rating, reviewText, reviewDate }) {
```
Comment (cmceachin5888) [inline db/models/reviews.js:3]: you don't need "id"

createReview({ productId, userId, rating, reviewText, reviewDate })

File: db/models/reviews.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    try {
+        const { rows } = await client.query(`
+        INSERT INTO reviews (id, ProductId, userId, rating, reviewText, reviewDate)
```
Comment (cmceachin5888) [inline db/models/reviews.js:6]: you dont need id

i think it should be (product_id, user_id, rating, review_text, review_date)

File: db/models/reviews.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        VALUES ($1, $2, $3, $4, $5, $6)
```
Comment (cmceachin5888) [inline db/models/reviews.js:7]: no need for $6

File: db/models/reviews.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        RETURNING *;
+        `,
+            [id, productId, userId, rating, reviewText, reviewDate]
```
Comment (cmceachin5888) [inline db/models/reviews.js:10]: [productId, userId, rating, reviewText, reviewDate]);

File: db/models/reviews.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (error) {
+        throw new Error('Could not create review: ' + error.message);
+    }
+}
+
+async function getAllReviews() {
+    try {
+        const { rows } = await client.query(`
+        SELECT *
+        FROM reviews
```
Comment (cmceachin5888) [inline db/models/reviews.js:23]: async function getAllReviews() {
  try {
    const { rows } = await client.query(`
      SELECT pr.id, pr.rating, pr.review_text, pr.review_date,
             u.name as reviewer_name, u.email as reviewer_email,
             p.name as product_name
      FROM product_reviews pr
      INNER JOIN users u ON pr.user_id = u.id
      INNER JOIN products p ON pr.product_id = p.id
    `);

    return rows;
  } catch (error) {
    throw new Error('Error fetching reviews: ' + error.message);
  }
}

File: db/models/reviews.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (error) {
+        throw new Error('Error fetching reviews: ' + error.message);
+    }
+}
+
+async function getReviewsByProduct(productId) {
+    try {
+        const { rows } = await client.query(`
+        SELECT * 
+        FROM reviews
```
Comment (cmceachin5888) [inline db/models/reviews.js:36]: async function getReviewsByProduct(productId) {
  try {
    const { rows } = await client.query(`
      SELECT pr.id, pr.rating, pr.review_text, pr.review_date,
             u.name as reviewer_name, u.email as reviewer_email
      FROM product_reviews pr
      INNER JOIN users u ON pr.user_id = u.id
      WHERE pr.product_id = $1
    `, [productId]);

    return rows;
  } catch (error) {
    throw new Error('Error fetching reviews for product: ' + error.message);
  }
}

File: db/models/reviews.js (hunk 5 above)
Comment (cmceachin5888) [inline db/models/reviews.js:6]: INSERT INTO product_reviews

**Review by cmceachin5888 — CHANGES_REQUESTED** [review]

**Review by cmceachin5888 — APPROVED** [review]
looks good

**Pull request merged**